### ISSUE TITLE
How to train M4C on InforgrahicVQA ?

### ISSUE THREAD

**Issue by Pxtri2156** [facebookresearch/mmf#1092]
## ❓ Questions and Help
Hi, everyone. I would like to train M4C on [InforgraphicVQA](https://arxiv.org/abs/2104.12756). However, I don't know what to build a dataset class for InforgraphicVQA. Moreover, I wonder if  I need to modify any configuration for training?  I have read the the guide line at [here](https://mmf.sh/docs), but I still do not understand how to train M4C on new datasets. Thereore, I hope, everyone can help me to solve that problem. 
Thank you for your attention.

**Comment by apsdehal** [comment]
@ronghanghu Can you help out here? Maybe point to your older comments on how to do this?

**Comment by ronghanghu** [comment]
Hi, one can run M4C on other datasets by building these datasets into imdbs and extracting their objects and OCR features. Basically, other datasets can be used by making them into similar formats as the TextVQA dataset. One can follow the steps in https://github.com/facebookresearch/mmf/issues/663#issuecomment-883000371.

**Comment by Pxtri2156** [comment]
@apsdehal @ronghanghu  I really appreciate you. I trained  M4C on InforgraphicVQA by following the steps in #663.

**Comment by soon22** [comment]
@ronghanghu Hi, is there a script to generate vocab file such as `fixed_answer_vocab_textvqa_5k.text` for TextVQA?

**Comment by ronghanghu** [comment]
Hi @soon22, I don't have a specific script to generate this vocab file, but it is basically tokenizing all the answers in the TextVQA dataset with a simple tokenizer and take the most frequent 5000 tokens, using the following tokenizer: https://github.com/facebookresearch/mmf/blob/3947693aafcc9cc2a16d7c1c5e1479bf0f88ed4b/mmf/utils/text.py#L64-L79

**Comment by soon22** [comment]
@ronghanghu Thanks for the response.
@Pxtri2156 Hi,
I would like to ask what PyTorch version and CUDA version you use to install `mmf` and `maskrcnn-benchmark`.
 I am trying to extract image features with https://github.com/facebookresearch/mmf/blob/master/tools/scripts/features/extract_features_vmb.py. I have installed `maskrcnn-benchmark`. I noticed when running `extract_features_vmb.py`, need to have`mmf` installed as well. I have try multiple combinations of CUDA and PyTorch
I am unable to install `mmf` and maskrcnn-benchmark` under the same version of CUDA and PyTorch and get them to work.
